Source/BoundaryConditions/WarpX_PEC.H

```diff
@@ -0,0 +1,205 @@
+     * \param[in] ijk_vec      indices along the x(i), y(j), z(k) of Efield Array4
+     * \param[in] n            index of the MultiFab component being updated
+     * \param[in] Efield       field data to be updated if (ijk) is at the boundary or a guard cell
+     * \param[in] stag         staggering of the field data being updated.
+     * \param[in] fbndry_lo    Field boundary type at the lower boundaries
+     * \param[in] fbndry_hi    Field boundary type at the upper boundaries
+     */
+    AMREX_GPU_DEVICE AMREX_FORCE_INLINE
+    void SetTangentialEfield (const int icomp, const amrex::IntVect & dom_lo,
+                                const amrex::IntVect &dom_hi,
+                                const amrex::IntVect &ijk_vec, const int n,
+                                amrex::Array4<amrex::Real> const& Efield,
+                                const amrex::IntVect& stag,
+                                amrex::GpuArray<int, 3> const& fbndry_lo,
+                                amrex::GpuArray<int, 3> const& fbndry_hi )
+    {
+        // Fields in guard cells set equal and opposite to cells in the mirror locations
+        // across the PEC boundary. Here we just initialize it.
+        amrex::IntVect ijk_mirror = ijk_vec;
+        int OnPECBoundary = 0;
+        int GuardCell = 0;
+        amrex::Real sign = 1._rt;
+#if (defined WARPX_DIM_XZ) || (defined WARPX_DIM_RZ)
+        const int sdim = ( (icomp == 1) ? 0 : 1 );
+#else
+        const int sdim = 1;
+#endif
+        // Loop over dimensions that are tangential to the Efield component, icomp.
+        // For 2D : for icomp==1, (Ey in XZ, Etheta in RZ), both X-Z and R-Z dimensions
+        //          are tangential. Therefore, we set sdim to 0 to loop over both
+        //          dimensions for this component.
+        // Otherwise : sdim = 1, to ensure we only loop over tangential dimensions.
+        for (int idim = sdim; idim < AMREX_SPACEDIM; ++idim) {
+            // dir is index of direction that is tangential to the component, icomp.
+            const int dir = ( ( (icomp+idim)<AMREX_SPACEDIM)
+                          ? (icomp+idim) : ( (icomp+idim) - AMREX_SPACEDIM));
+            // loop over sides, iside = 0 (lo), iside = 1 (hi)
+            for (int iside = 0; iside < 2; ++iside) {
+                const int is_tangent_to_PEC = ( ( iside == 0 )
+                                            ? is_boundary_PEC(fbndry_lo, dir)
+                                            : is_boundary_PEC(fbndry_hi, dir) );
+                if (is_tangent_to_PEC == 1) {
+                    // guard cell outside the domain by "ig" number cells, in direction, dir
+                    const int ig = ( (iside == 0)
+                                 ? (dom_lo[dir] - ijk_vec[dir])
+                                 : (ijk_vec[dir] - (dom_hi[dir] + stag[dir]) ) );
+
+                    if ( ig == 0) OnPECBoundary = 1;
```

Comment:
Can this be changed to `if (ig == 0) {} else if (ig > 0) {}`? This would be more clear and slightly faster.